component/AuthenticatedLayout.tsx

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import React, { useEffect, useState } from 'react';
+import { verifyToken } from '@/services/validation';
+import { useRouter } from 'next/navigation';
+
+interface AuthenticatedLayoutProps {
+    children: React.ReactNode;
+}
+
+function AuthenticatedLayout({ children }: AuthenticatedLayoutProps) {
+    const router = useRouter();
+    const [verificationResult, setVerificationResult] = useState<{
+        isValid: boolean;
+        message: string;
+        data?: any;
+    }>({
+        isValid: false,
+        message: 'Verifying token...',
+    });
+
+    useEffect(() => {
+        async function checkToken() {
+            try {
+                const result = await verifyToken();
+                setVerificationResult({
+                    isValid: result.isValid,
+                    message: result.message || 'No message available',
+                    data: result.data,
+                });
+
+                if (!result.isValid) {
+                    router.back();
```

Comment:
1. We should set state for the whole application here. States can be `loggedOut`, `loggedIn` (with User ID as part of the state)
2. Gurukul will have an anonymous library that can be browsed even when not logged in. So: create a page that will also show "Welcome to Gurukul", and a button that says "Log in" that will redirect to the portal
3. Don't do `router.back()` -- put an explicit redirect to portal here